coderd/database/queries/insights.sql

```diff
@@ -771,3 +771,18 @@ SELECT
 FROM unique_template_params utp
 JOIN workspace_build_parameters wbp ON (utp.workspace_build_ids @> ARRAY[wbp.workspace_build_id] AND utp.name = wbp.name)
 GROUP BY utp.num, utp.template_ids, utp.name, utp.type, utp.display_name, utp.description, utp.options, wbp.value;
+
+-- name: GetAccumulatedUsersInsights :many
+-- GetAccumulatedUsersInsights returns the accumulated number of users created
+-- in the given timeframe. It returns the accumulated number of users for each date
+-- within the specified timeframe, providing a running total of user sign-ups.
+SELECT
+	date_trunc('day', created_at)::date AS date,
```

Comment:
This might produce strange results if requested start time and end time are in a different timezone. Each day may be missing few/many entries and we might get ghost dates as well (+/-1 day outside start/end range).

Suggestion:
```sql
	date_trunc('day', (
		created_at AT TIME ZONE EXTRACT(TIMEZONE FROM @start_time::timestamptz)
	)) AT TIME ZONE EXTRACT(TIMEZONE FROM @start_time::timestamptz) AS date,
```

(Untested), why the double at timezone? One strips timezone and the other one adds it back.

```
coder=# select created_at from users;
          created_at
-------------------------------
 2024-04-08 11:41:35.096834+00
 2024-04-08 11:40:54.587231+00
(2 rows)

coder=# select created_at at time zone 'europe/helsinki' from users;
          timezone
----------------------------
 2024-04-08 14:41:35.096834
 2024-04-08 14:40:54.587231
(2 rows)

coder=# select created_at at time zone 'europe/helsinki' at time zone 'europe/helsinki' from users;
           timezone
-------------------------------
 2024-04-08 11:41:35.096834+00
 2024-04-08 11:40:54.587231+00
(2 rows)
```

Notice +00 and lack thereof. Now notice how we get a timezoneally correct truncation when we do it without timezone:

```
coder=# select date_trunc('day', created_at) from users;
       date_trunc
------------------------
 2024-04-08 00:00:00+00
 2024-04-08 00:00:00+00
(2 rows)

coder=# select date_trunc('day', created_at at time zone 'europe/helsinki') at time zone 'europe/helsinki' from users;
        timezone
------------------------
 2024-04-07 21:00:00+00
 2024-04-07 21:00:00+00
(2 rows)
```

You might want to use a subquery so you can count over the improved truncation without repeating all that noise.